src/tools/tidy/src/x_version.rs

```diff
@@ -0,0 +1,52 @@
+use semver::{BuildMetadata, Prerelease, Version};
+use std::io::ErrorKind;
+use std::process::{Command, Stdio};
+
+pub fn check(bad: &mut bool) {
+    let result = Command::new("x").arg("--wrapper-version").stdout(Stdio::piped()).spawn();
+    // This runs the command inside a temporary directory.
+    // This allows us to compare output of result to see if `--wrapper-version` is not a recognized argument to x.
+    let temp_result = Command::new("x")
+        .arg("--wrapper-version")
+        .current_dir(std::env::temp_dir())
+        .stdout(Stdio::piped())
+        .spawn();
+
+    let (child, temp_child) = match (result, temp_result) {
+        (Ok(child), Ok(temp_child)) => (child, temp_child),
+        (Err(e), _) | (_, Err(e)) => match e.kind() {
+            ErrorKind::NotFound => return,
+            _ => return tidy_error!(bad, "failed to run `x`: {}", e),
+        },
+    };
+
+    let output = child.wait_with_output().unwrap();
+    let temp_output = temp_child.wait_with_output().unwrap();
+
+    if output != temp_output {
+        return tidy_error!(
+            bad,
+            "Current version of x does not support the `--wrapper-version` argument\nConsider updating to the newer version of x by running `cargo install --path src/tools/x`"
+        );
+    }
+
+    if output.status.success() {
+        let version = String::from_utf8_lossy(&output.stdout);
+        let version = Version::parse(version.trim_end()).unwrap();
+        let expected = Version {
+            major: 0,
+            minor: 1,
+            patch: 0,
+            pre: Prerelease::new("").unwrap(),
+            build: BuildMetadata::EMPTY,
+        };
```

Comment:
I'm worried people will forget to update this when they change the version in `src/tools/tidy`. Can you use the rust equivalent of `cargo metadata --no-deps --manifest-path src/tools/x/Cargo.toml | jq '.packages | select(.name == "x") | .version'` to choose the expected version instead?